doc/reference/reference_sql/sql_user_guide.rst

```diff
@@ -945,50 +968,54 @@ So AAA = Always for explicit, Always for Implicit (assignment), Always for Impli
 
 The S "Sometimes allowed" character applies for these special situations: |br|
 From STRING To BOOLEAN is allowed if UPPER(string-value) = ``'TRUE'`` or ``'FALSE'``. |br|
-From number to INTEGER or UNSIGNED is allowed for cast and assignment only if the result is not out of range,
-and the number has no post-decimal digits. |br|
-From STRING to INTEGER or UNSIGNED is allowed only if the string has a representation of a number,
+From numeric to INTEGER or UNSIGNED is allowed for cast and assignment only if the result is not out of range,
+and the numeric has no post-decimal digits. |br|
+From STRING to INTEGER or UNSIGNED or DECIMAL is allowed only if the string has a representation of a numeric,
 and the result is not out of range,
-and the number has no post-decimal digits. |br|
-From STRING to DOUBLE or NUMBER is allowed only if the string has a representation of a number. |br|
-From BOOLEAN to number is allowed only if the number is not DOUBLE. |br|
+and the numeric has no post-decimal digits. |br|
+From STRING to DOUBLE or NUMBER is allowed only if the string has a representation of a numeric. |br|
 From STRING to UUID is allowed only if the value is
 (8 hexadecimal digits) hyphen (4 hexadecimal digits) hyphen (4 hexadecimal digits) hyphen (4 hexadecimal digits) hyphen (12 hexadecimal digits),
 such as ``'8e3b281b-78ad-4410-bfe9-54806a586a90'``. |br|
 From VARBINARY to UUID is allowed only if the value is
 16 bytes long,
 as in ``X'8e3b281b78ad4410bfe954806a586a90'``. |br|
 The chart does not show To|From SCALAR because the conversions depend on the type of the value,
-not the type of the column definition.
-Explicit cast to SCALAR is allowed but has no effect, the result data type is always the same as the original data type.
-But comparisons of values of different types are allowed if the definition is SCALAR.
+not the type of the column definition.  Explicit cast to SCALAR is always allowed.
 
 ..  note::
 
     Since version :doc:`2.4.1 </release/2.4.1>`, the NUMBER type is processed
     in the same way as the :ref:`number <index-box_number>` type in
     NoSQL Tarantool.
 
+    Starting with Tarantool 2.10.1, these conversions which used to be legal are now illegal: |br|
+    Explicit cast from numeric to BOOLEAN, |br|
+    Explicit cast from BOOLEAN to numeric, |br|
+    Implicit cast from NUMBER to other numeric types for arithmetic or built-in functions. |br|
+    Implicit cast from numeric to STRING. |br|
+    Implicit cast from STRING to numeric.
+
 Examples of casts, illustrating the situations in the chart:
 
-``CAST(TRUE AS INTEGER)`` is legal because the intersection of the  "From BOOLEAN" row with the "To number"
+``CAST(TRUE AS STRING)`` is legal because the intersection of the  "From BOOLEAN" row with the "To STRING"
```

Comment:
```suggestion
``CAST(TRUE AS STRING)`` is legal because the intersection of the "From BOOLEAN" row with the "To STRING"
```

Reply:
@pgulutzan also it sounds now like this cast is legal because of the contents of the mentioned table, which is not actually true. :) Should we maybe rephrase it to make it clear that this cast is allowed because of the rules of casting or underlying algorithms in Tarantool? Or just leave this part out at all, and rephrase it like "``CAST(TRUE AS STRING)`` is legal; you can see that in the intersection of  "From BOOLEAN" row with the "To STRING" column" and so on.

Reply:
@alexandra-mara: You are right, but "CAST(TRUE AS STRING) is legal; you can see that in the intersection of the  From BOOLEAN row with the To STRING column is A--" etc. would not be perfectly grammatical, so I would suggest "CAST(TRUE AS STRING) is legal. The intersection of the From BOOLEAN row with the To STRING column is A--" etc. I can make that change in each example, and fix the spaces, and push again. Okay?

Reply:
@pgulutzan, yeah, sure

Reply:
Okay, changed in latest commit.